### PR TITLE
docs: Update dev guide for OBS packages

### LINKED ISSUE #80: Update dev guide for OBS installation packages
Add details to the developer guide explaining that the recommended way to install Kata is using the OBS binary packages.

Devs can then override particular components by building them from source.

### PR DESCRIPTION
**Pull request by jodh-intel**
Add details of how to install the packaged components to start off with
a working system. This involved splitting out part of the "Assumptions"
section into a new "Initial setup" section.

Fixes #80.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Hi @klynnrif - ptal.

File: Developer-Guide.md [hunk 1]
```diff
@@ -27,23 +29,32 @@
...
-
-    https://github.com/clearcontainers/runtime/wiki/Installation
+
+# Initial setup
+
+The recommended way to create a development environment is to first install the
+packaged versions of the Kata Containers components to create a working
+system:
+
+  * [Fedora*](install/fedora-installation-guide.md)
```
Comment (egernst) [inline Developer-Guide.md]: Will this create a mix of /usr/bin and /usr/local/bin on the system, though?  Should we just include directions on installing docker, qemu and VM artifacts?

File: Developer-Guide.md [hunk 2]
```diff
@@ -27,23 +29,32 @@
...
+  * [Ubuntu*](install/ubuntu-installation-guide.md)
+
+The installation guide instructions will install all required Kata Containers
+components, plus Docker*, the hypervisor, and the Kata Containers image and
```
Comment (egernst) [inline Developer-Guide.md]: In general, I think all of our documentation assumes that all someone will ever want to do is setup Kata with Docker.  Its a good step, but I think we should be a bit more general, or at least include better install directions for Kubernetes development (this is an aside -- nothing we should address in this PR...).

File: Developer-Guide.md (hunk 1 above)
Comment (jodh-intel) [reply Developer-Guide.md]: Yes - the official OBS binary for the runtime will be `/usr/bin/kata-runtime` and the developer version will be `/usr/local/bin/kata-runtime`. That's documented in the guide already (see https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md#update-docker-configuration). We really don't want to encourage users to overwrite the packaged binaries, even on a dev box imho.

File: Developer-Guide.md (hunk 2 above)
Comment (jodh-intel) [reply Developer-Guide.md]: Agreed. I think that's a post release concern ;)

**Comment by jodh-intel** [comment]
Hi @klynnrif, @egernst et al - I've updated the branch so please could you take a fresh look? :)

**Pull request merged**